phasespace/fromdecay/genmultidecay.py

```diff
@@ -0,0 +1,290 @@
+import itertools
+from typing import Callable, Union
+
+import tensorflow as tf
+import tensorflow.experimental.numpy as tnp
+from particle import Particle
+
+from phasespace import GenParticle
+
+from .mass_functions import _DEFAULT_CONVERTER
+
+_MASS_WIDTH_TOLERANCE = 0.01
+_DEFAULT_MASS_FUNC = "relbw"
+
+
+class GenMultiDecay:
+    def __init__(self, gen_particles: list[tuple[float, GenParticle]]):
+        """A container that works like GenParticle that can handle multiple decays. Can be created from.
+
+        Args:
+        gen_particles: All the GenParticles and their corresponding probabilities.
+            The list must be of the format [[probability, GenParticle instance], [probability, ...
+        """
+        self.gen_particles = gen_particles
+
+    @classmethod
+    def from_dict(
+        cls,
+        dec_dict: dict,
+        mass_converter: dict[str, Callable] = None,
+        tolerance: float = _MASS_WIDTH_TOLERANCE,
+    ):
+        """Create a GenMultiDecay instance from a dict in the decaylanguage format.
+
+        Args:
+            dec_dict:
+                The input dict from which the GenMultiDecay object will be created from.

```

Comment:
@jonas-eschle, are you using cross-referencing in your docs (https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#cross-referencing-python-objects), in which case instances such as GenMultiDecay should be cross-referenced.

Reply:
Yes, indeed, we use it.

Reply:
Happy also to have this follow-up in https://github.com/zfit/phasespace/issues/68.